core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -1086,11 +1086,18 @@ private[spark] class TaskSchedulerImpl(
     case ExecutorKilled =>
       logInfo(s"Executor $executorId on $hostPort killed by driver.")
     case _: ExecutorDecommission =>
-      logInfo(s"Executor $executorId on $hostPort is decommissioned.")
+      logInfo(s"Executor $executorId on $hostPort is decommissioned after " +
+        s"${Utils.msDurationToString(getDecommissionDurationMs(executorId))}.")
     case _ =>
       logError(s"Lost executor $executorId on $hostPort: $reason")
   }
 
+  private def getDecommissionDurationMs(executorId: String): Long = {
+    executorsPendingDecommission.get(executorId)
+      .map(s => clock.getTimeMillis() - s.startTime)
+      .getOrElse(0L)
```

Comment:
Could you add a function description. Especially, describing about what is the meaning of `0L` when `executorsPendingDecommission.get` doesn't exist.